### DISCUSSION TITLE
Add ability to copy/paste node signals to other nodes.

### DISCUSSION THREAD

**Discussion by yosimba2000** [godotengine/godot-proposals#7738] [Editor]
Sometimes you have multiple nodes that all need the same exact Signals. Currently, you have to go thru each Node, find the right signal, find the right method from your script, and then input the right argument for that method.

This is very tedious, and would be much faster if we could set up the Signal for one node, then click Copy Signal from a context menu, then pasting that Signal onto a new node from the context menu again.

Here's a simple use case:
You have 10 UI buttons, and you want to assign a "button hover" sound when mouse enters the button. You would have to assign the Signals 10 times, one for each button. Instead, it would be easier to set up the Signal once for a button, then copy/paste to all other buttons.

**Comment by SeaLiteral** [comment]
If it's specifically for buttons, I think you could probably create one, duplicate it a bunch of times, and then have several buttons with the same signal connection. I guess if they aren't in something like a `BoxContainer` they'd end up all in the same place, but be draggable to other places.

That being said, if instead of several buttons you were trying to add one signal handler to several nodes of different types, duplicating doesn't seem to fix that. And while one could still add those signal connections manually a bunch of times, I do think being able to copy-paste them would save Godot users more time than I think it would take Godot developers to implement it.

**Reply by yosimba2000** [reply]
This would be for all nodes in general, only if the source and destination signal types match of course.

I think I tried that, but the signals didn't completely copy over.

But even if they did, without a copy/paste signals ability, it would not be conducive to an iterative design workflow, which of course gamedev is.

You'd have to know exactly which buttons you want and what they do on the first try if the only way to copy signals is to copy/paste nodes. Otherwise you end up with the same problem if you want to adjust the signals. You'd have to adjust the signals then for all nodes, or delete the copied nodes, and duplicate them and rearrange them again.

**Comment by azur-wolve** [comment]
a trait system is on the milestone
it would allow a similar functionality as C# interfaces but even more flexibility

that would allow for that
